code/__HELPERS/math/vector/vector.dm

```diff
@@ -0,0 +1,141 @@
+var/list/vector/cardinal_vectors = list(
+	new /vector(0,1), //NORTH
+	new /vector(0,-1), //SOUTH
+	new /vector(1,0), //EAST
+	new /vector(-1,0) //WEST
+)
+var/list/vector/diagonal_vectors = list(
+	new /vector(1,1), //NORTHEAST
+	new /vector(-1,1), //NORTHWEST
+	new /vector(1,-1), //SOUTHEAST
+	new /vector(-1,-1) //SOUTHWEST
+)
+var/list/vector/all_vectors = list(
+	new /vector(0,1), //NORTH
+	new /vector(0,-1), //SOUTH
+	new /vector(1,0), //EAST
+	new /vector(-1,0), //WEST
+	new /vector(1,1), //NORTHEAST
+	new /vector(-1,1), //NORTHWEST
+	new /vector(1,-1), //SOUTHEAST
+	new /vector(-1,-1) //SOUTHWEST
+)
+
+// Basic geometry things.
+/vector
+	var/x = 0
+	var/y = 0
+
+/vector/New(var/x, var/y)
+	src.x = x
+	src.y = y
+
+/vector/proc/duplicate()
+	return new /vector(x, y)
+
+/vector/proc/euclidian_norm()
+	return sqrt(x*x + y*y)
+
+/vector/proc/squared_norm()
+	return x*x + y*y
+
+/vector/proc/normalized()
+	var/norm = euclidian_norm()
+	return new /vector(x/norm, y/norm)
+
+/vector/proc/floored()
+	return new /vector(Floor(x), Floor(y))
+
+//use this one
+/vector/proc/chebyshev_norm()
+	return max(abs(x), abs(y))
+
+//use this one
+/vector/proc/chebyshev_normalized()
+	var/norm = chebyshev_norm()
+	return new /vector(x/norm, y/norm)
+
+/vector/proc/is_integer()
+	return IS_INT(x) && IS_INT(y)
+
+/vector/proc/is_null()
+	return chebyshev_norm() == 0
+
+/vector/proc/toString()
+	return "\[Vector\]([x],[y])"
+
+//returns angle from 0 to 360
+//-1 if vector is (0,0)
+//angle calculated on north
+/vector/proc/toAngle()
+	if(x == 0)
+		if(y == 0)
+			return -1
+		else if(y > 0)
+			return 0
+		else if(y < 0)
+			return 180
+	else if(y == 0)
+		if(x > 0)
+			return 90
+		else if(x < 0)
+			return 270
+
+	var/vector/src_norm = src.chebyshev_normalized()
+	var/angle = arctan(src_norm.y,src_norm.x) - 360 * -1 //this is broken
```

Comment:
2D angle for a nonzero (x, y) pair of euclidean coordinates is `arctan(y/x)`, with 0 on the Ox positive axis (effectively counted from the EAST dir).

Just subtract 90 degrees if you want them counted from the north.

Reply:
if i substract 90 degrees it will count still count conter-clockwise tho. i thought it counting clockwise would be much more sensible